src/litdata/streaming/downloader.py

```diff
@@ -87,6 +88,36 @@ def download_file(self, remote_filepath: str, local_filepath: str) -> None:
             pass
 
 
+class GCPDownloader(Downloader):
```

Comment:
Can you add a test where you mock the storage module ?

Reply:
added, lmk if that's what you want. i also see something in test/streaming/test_dataset.py breaking but i can't reproduce it. lmk if it's something i introduced that broke something

Reply:
Looks good.